SYCL/Basic/in_order_queue_status.cpp

```diff
@@ -0,0 +1,84 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// Test checks that queue::ext_oneapi_empty() returns status of the in-order
+// queue.
+
+#include <chrono>
+#include <iostream>
+#include <sycl.hpp>
+#include <thread>
+
+static void CheckArray(int *x, size_t buffer_size, int expected) {
+  for (size_t i = 0; i < buffer_size; ++i) {
+    assert(x[i] == expected);
+  }
+}
+
+using namespace sycl;
+
+void TestFunc(queue &Q) {
+  static constexpr int Size = 100;
+
+  assert(Q.ext_oneapi_empty() && "Queue is expected to be empty");
+
+  int *X = malloc_host<int>(Size, Q);
+  int *Y = malloc_host<int>(Size, Q);
+
+  auto FillEv = Q.fill(X, 99, Size);
+  auto SingleTaskEv = Q.submit([&](handler &CGH) {
+    auto SingleTask = [=] {
+      for (int I = 0; I < Size; I++)
+        X[I] += 1;
+    };
+    CGH.single_task(SingleTask);
+  });
+  auto MemCpyEv = Q.copy(X, Y, Size);
+  constexpr int NumIter = 5;
+  for (int I = 0; I < NumIter; I++) {
+    Q.submit([&](handler &CGH) {
+      CGH.parallel_for<class Kernel1>(sycl::range<1>(Size),
+                                      [=](sycl::id<1> WI) { Y[WI] *= 2; });
+    });
+  }
+
+  // Wait a bit to give a chance for tasks to complete.
+  std::this_thread::sleep_for(std::chrono::milliseconds(500));
```

Comment:
could we try to avoid sleeps in tests? we could not rely on such approach.
what if to add host_task as the last task that would signal cv or lock/unlock mutex and simply wait for it instead of sleep, will it help?

Reply:
> what if to add host_task as the last task that would signal cv, atomic or lock/unlock mutex and simply wait for it instead of sleep, will it help?

Unfortunately, as far as I know SYCL doesn't guarantee that such waiting will ever complete. User must perform synchronization to guarantee completion. I mean it is not guaranteed that queue will be flushed and we may just wait forever.

Reply:
@KseniyaTikhomirova, unfortunately, I don't have good ideas how to get rid of sleep right now for the reasons I described above. I can probably replace it with some loop - try several times to see if queue is empty and give up if it is not empty - but I'm not sure whether it is better or not.

Reply:
ok, unfortunately I also do not have better ideas about how to get rid of sleep call. If we want to check that the right value could be returned even without wait call - may be it could be done with unittest. Although I do not want to block you here.

Reply:
Thank you! Unittest may be a good idea.